Resolve theta build/run issues

bgeltz opened the pull request:
- Fixes #1924, #1923.

bgeltz committed:
Add theta env

Change-Id: Ib6334b9bd41dea0fe3c3e33680ec5a669c944d8d
Signed-off-by: Brad Geltz <brad.geltz@intel.com>

bgeltz committed:
Run compute commands with quiet flag

- Extends quiet funcionality to suppress aprun
  messages.

Change-Id: I5aa88a3774ac2411f0a877ea5a127969537a9047
Signed-off-by: Brad Geltz <brad.geltz@intel.com>

bgeltz committed:
Quote LD_PRELOAD properly for aprun

Change-Id: Ifcbbd5b76282536fe4c83beff026a05f8bc5a16c
Signed-off-by: Brad Geltz <brad.geltz@intel.com>

bgeltz committed:
Move compiler overrides out of make lines

- Avoids parameter expansion issues when CC is
  set to 'cc -dynamic'.

Change-Id: I79c190609af5b087dd809ded312a60a8a36c5daa
Signed-off-by: Brad Geltz <brad.geltz@intel.com>

cmcantalupo commented on integration/config/build.sh:
```diff
@@ -128,7 +128,7 @@ build(){
 
 # Run the service build
 cd service
-CC=gcc CXX=g++ build "${GEOPM_SERVICE_CONFIG_OPTIONS}"
```
I wonder if we need this change in geopm-service.spec.in?

cmcantalupo commented on scripts/geopmpy/launcher.py:
```diff
@@ -992,7 +993,7 @@ def launcher_command(self):
         Returns 'srun', the name of the SLURM MPI job launch application.
         """
-        return 'srun'
+        return ['srun']
```
It is not very intuitive that a method that is called "launcher_command" should return a list of strings with one element. At the very least, all of the documentation for these methods would have to change to describe the return type as a list, but I'm really not sure why the caller can't be responsible for creating the list that will become argv.  

The way this is resolved for all of the other launcher command line options is to add a new method to the launcher base that returns empty list e.g. quiet_option() and then for the AprunLauncher this would be implemented as return ['-q'].  I'm not saying the launcher class is beautiful, but that is how it works.

cmcantalupo reviewed (changes requested):
Thanks for the debug.  I know using Popen without a shell interpreter is pretty confusing, and that is the root cause of the LD_PRELOAD issue.  

I know the launcher is far from perfect, but I'll like to keep the implementation at least consistent.  Please add a new method to the launcher base class.  Thanks!

bgeltz committed:
Clear <C|CXX>FLAGS when building the service RPM

- Handles cases where they were set to something in the
  calling environment that is not appropriate for the
  GNU build.  (e.g. theta_env.sh sets the AVX flag)

Signed-off-by: Brad Geltz <brad.geltz@intel.com>

bgeltz committed:
Move quiet option into its own method

Signed-off-by: Brad Geltz <brad.geltz@intel.com>
